### PR TITLE
Chore/react native example

### PR DESCRIPTION
**Pull request by saurabhsri108**
## What kind of change does this PR introduce?

Chore #4702 

## What is the current behavior?

It's adding a new folder inside an example named ReactNativeTodoList done on react-native 0.66, gradle version 7.3, Android SDK version 30.0.3 and JDK version 17.0.1. No TypeScript used currently.

## What is the new behavior?
- [x] Create a new folder in example folder.
- [x] Create simple TodoList app using latest versions of react-native, gradle, android sdk, jdk.
- [x] Test the app and check for errors
- [x] Update README.md file to showcase the exact steps for installing latest gradle and jdk as well as setting up the SUPABASE database.
- [x] Add a new typescript version for the same.
- [ ] Test the ios build

## Additional context
Someone needs to test the ios build if its necessary for the example. Don't have Mac

### CODE REVIEW

**Comment by vasudeveloper001** [comment]
Can i get assigned to this one @kiwicopple ? I would love to get feedback if people have free time.

File: examples/ReactNativeTodoList/.gitignore [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+*.perspectivev3
+!default.perspectivev3
+xcuserdata
+*.xccheckout
+*.moved-aside
+DerivedData
+*.hmap
+*.ipa
+*.xcuserstate
+
```
Comment (Isaiah-Hamilton) [inline examples/ReactNativeTodoList/.gitignore:23]: Not really sure if I did this right but should add `.vscode` to remove personal settings.

```suggestion

# .vscode
.vscode

```

**Review by Isaiah-Hamilton — COMMENTED** [review]
Add a prettier ignore just like `examples/ReactNativeTodoList/.prettierrc.js`.

File: examples/ReactNativeTodoList/.gitignore (hunk 1 above)
Comment (vasudeveloper001) [reply examples/ReactNativeTodoList/.gitignore:23]: Added the .vscode, .idea, and .prettierjs.js names in the gitignore files of both the 2 example folders I created.

**Review by kiwicopple — APPROVED** [review]

**Comment by kiwicopple** [comment]
Thanks for this @vasudeveloper001 - and sorry for the delay. I was travelling.

This is looking great, although admittedly I don't have a huge amount of RN experience. I think it's looking good to go though - I'm sure other contributors can improve it if they find any necessary changes.

Feel free to email swag@supabase.io - this is definitely worthy of a tshirt and some stickers :).Make sure to include a link to this PR!

**Comment by vasudeveloper001** [comment]
Thanks @kiwicopple for accepting the pull request and merging. This was my first ever open source contribution. Loved it and will be doing more on Supabase as I learn more things in Software Engineering and JavaScript. 

Should I delete the branch now? I think that's the right way to do the finishing touch after successful merge.

PS: And I have sent the email to the mail link above.

**Comment by kiwicopple** [comment]
> Should I delete the branch now

Yes! you can if you want to

> This was my first ever open source contribution. 

wow - this is an amazing first contribution. THanks for this @vasudeveloper001 - a great addition to the community

**Pull request merged**